lib/src/repo_path.rs

```diff
@@ -76,6 +78,28 @@ impl RepoPath {
         RepoPath { components }
     }
 
+    /// Parses an `input` path into a `RepoPath` relative to `base`.
+    ///
+    /// The `cwd` and `base` paths are supposed to be absolute and normalized in
+    /// the same manner. The `input` path may be either relative to `cwd` or
+    /// absolute.
+    pub fn parse_fs_path(cwd: &Path, base: &Path, input: &str) -> Result<Self, FsPathParseError> {
+        use std::path::Component;
```

Comment:
Can this be at the top of the file? It doesn't look like it would conflict with anything.

Reply:
sure, done.